### DISCUSSION TITLE
Strategy to migrate and test GH Actions pipelines

### DISCUSSION THREAD

**Discussion by leandroberetta** [kiali/kiali#4994] [General]
Hello, regarding the epic: #4827, we have two PRs for the migrated Jenkins pipelines to GH Actions, the Kiali server and the Kiali operator pipelines.

- https://github.com/kiali/kiali/pull/4980
- https://github.com/kiali/kiali-operator/pull/520

Between today and tomorrow I will be preparing the PRs for the Helm charts and the site. I'm trying to make the 4 pipelines very similar in terms of jobs and steps for the release pipeline, so these upcoming pipelines will be faster to do than the ones that already have a pull request.

This message is to propose an strategy to start usem them and most importantly, to test them. 

I think that for the upcoming release we might need to keep the releases made by Jenkins and also spin the GH Actions ones, something very similar to a Dark release. Then we can test the images and see if anything is properly created (although I tested all the steps in my own repositoy and Quay repository I think this is required to be sure, more eyes the better).

If this strategy is accepted, we can merge the PRs as for today, the only way to start a GH actions release pipeline is manually, and in that action we can input some parameters to control the outputs:

![pipeline](https://user-images.githubusercontent.com/1286393/165095050-2b778a15-b079-4234-a1bc-8ba1c16e64e7.png)

**Comment by lucasponce** [comment]
Thanks, @leandroberetta.

I think it depends on how you see this work, if it's stable, to me it's ok to try to release upstream via the new Github actions pipeline and leave Jenkins as a backup in case something goes wrong.

But you decide, I have no strong opinion.

**Reply by jshaughn** [reply]
Because we don't have a firm commitment for upstream release timing, if it is easier to test and refine the gh pipeline with real releases I am not opposed to just making the switch.  We could also generate more alpha pre-releases this week, if that makes sense.  We already have a 1.50-alpha0 and 1.50-alpha1, we could try an alpha 2,3,4 etc.

**Reply by israel-hdez** [reply]
It is worth to mention that alpha releases are not making commits to the `master` branches of the repos. Also, in quay, the `vX.Y` tag that mirrors the full `vX.Y.Z` tag is not created. So, it's just to note that even when doing alpha releases, some things will remain untested until the minor release is done.

I would defer to @leandroberetta to tell if he is confident with the switch. Perhaps, at minimim, and if everybody is OK, we could do the release in the America's morning to make sure that all people is online if something needs to be done manually afterwards.

**Comment by israel-hdez** [comment]
> If this strategy is accepted, we can merge the PRs as for today, the only way to start a GH actions release pipeline is manually, and in that action we can input some parameters to control the outputs:
> 
> ![pipeline](https://user-images.githubusercontent.com/1286393/165095050-2b778a15-b079-4234-a1bc-8ba1c16e64e7.png)

Can you, please, update the [RELEASING.adoc](https://github.com/kiali/kiali/blob/master/RELEASING.adoc) with the new instructions to generate releases? This file are just minimal comments/instructions for creating the releases. Later, and once everything is more stable, I hope we can have a more complete document [like the one I have for Jenkins](https://github.com/kiali/kiali/tree/master/deploy/jenkins-ci).

**Reply by leandroberetta** [reply]
Sure, thanks for the heads up.
